### ISSUE TITLE
OUSD compensation claiming: two txs issued in the same block

### ISSUE THREAD

**Issue by franckc** [OriginProtocol/origin-dollar#512]
Looking at the [transactions](https://etherscan.io/address/0x501804b374ef06fa9c427476147ac09f1551b9a0) for the OUSD compensation contract, I noticed that the wallet 0x495875C49f7BF572B2A5EA6F82E86EE43E462f8d created 2 transactions to claim OUSD in the same block 11690893:

tx hash 1: https://etherscan.io/tx/0x197bffccedee2ebd3222de5f1656af5130bdf92c76b2c966a8c973c3e6d85b17
tx hash 2: https://etherscan.io/tx/0x940e3d4838353e28a527622fd47408521be9371159be6048d35b958a31b858d4

As expected the second transaction fails since the first transaction already claimed the OUSD.

But I wonder if this could be a bug in our DApp that would issue 2 transactions?

**Comment by franckc** [comment]
@sparrowDom Would it be a possible bug in the DApp?

**Comment by sparrowDom** [comment]
Found a solution for the above bug here: 
https://github.com/OriginProtocol/origin-dollar/commit/9994c70b4b236fa9188b22b64268aa5d3f1ecdf1

It is kind of a weird thing. In forked environment I connected a `compensationClaims.balanceOf` call to a button and kept clicking it. Observing the node:fork output I could see that the call reached the node for the first time, but not subsequent clicks. That makes me think Metamask has some sort of caching implemented and doesn't pass on the calls to its provider immediately (maybe to save on the api calls?). Or uses event subscription with its provider... I don't know.

A working solution (I am not super happy with) is just querying the provider each second after the call until an expected 0 balance shows up.
